app/gui2/shared/ast/tree.ts

```diff
@@ -1398,17 +1378,27 @@ export class MutableTextLiteral extends TextLiteral implements MutableAst {
     this.fields.set('close', unspaced(Token.new(code)))
   }
 
-  setTextContents(rawText: string) {
-    const isInterpolated = this.isInterpolated()
-    const token = this.boundaryTokenVariant()
-    const mustBecomeInterpolated = !isInterpolated && (!token || rawText.includes(token))
-    const doEscape = isInterpolated || mustBecomeInterpolated
-
-    if (mustBecomeInterpolated) this.setBoundaries(token === '"""' ? "'''" : "'")
+  setElements(elements: TextElement<OwnedRefs>[]) {
+    this.fields.set(
+      'elements',
+      elements.map((e) => mapRefs(e, ownedToRaw(this.module, this.id))),
+    )
+  }
 
-    const codeText = doEscape ? escapeInterpolation(rawText) : rawText
-    const newElement = concreteChild(this.module, unspaced(Token.new(codeText)), this.id)
-    this.fields.set('elements', [newElement])
+  setContentUninterpolated(rawText: string) {
+    let boundary = this.boundaryTokenCode()
+    const isInterpolated = this.isInterpolated()
+    const mustBecomeInterpolated = !isInterpolated && (!boundary || rawText.includes(boundary))
+    if (mustBecomeInterpolated) {
+      boundary = "'"
+      this.setBoundaries(boundary)
+    }
+    const literalContents =
+      isInterpolated || mustBecomeInterpolated ? escapeInterpolation(rawText) : rawText
```

Comment:
If the text is uninterpolated (so, it is already escaped), why do we escape it again?

Reply:
Oh, it means "not meant to be interpolated" not "not yet interpolated".